src/util/StringUtils.h

```diff
@@ -597,35 +224,6 @@ inline string strip(const string& text, const T& s) {
   return text.substr(pos, length);
 }
 
-inline size_t findClosingBracket(const string& haystack, size_t start,
-                                 char openingBracket, char closingBracket) {
-  if (haystack[start] != openingBracket) {
-    throw ParseException(
-        "The string " + haystack + " does not have a opening bracket " +
-        string(1, openingBracket) + " at position " + std::to_string(start));
-  }
-  int depth = 0;
-  size_t i;
-  for (i = start + 1; i < haystack.size(); i++) {
-    if (haystack[i] == openingBracket) {
-      depth++;
-    }
-    if (haystack[i] == closingBracket) {
-      if (depth == 0) {
-        return i;
-      } else {
-        depth--;
-      }
-    }
-  }
-  if (depth == 0) {
-    throw ParseException("Need curly braces in string " + haystack + "clause.");
-  } else {
-    throw ParseException("Unbalanced curly braces.");
-  }
-  return -1;
-}
-
```

Comment:
Were all these functions completely unused in the current QLever codebase?

I think this utils files was copied over from a previous project. I think most of them became unused,
when we refactored the parsing to use (more) proper and generic approaches.

I am not sure whether we should leave `utf8ToUpper` in as the "dual" to the lowercasing, but then again, we don't
really need it, and it can be reimplemented easily, so we should probably remove it.

Reply:
Same with `lstrip`, `strip` and `rstrip` are used but I don't think it's worth to keep them, especially because it looks like absl already has those kind of functions:
https://github.com/abseil/abseil-cpp/blob/e3fdd9b16a2a90c9e01e00de46605ce59bebc661/absl/strings/ascii.h#L230-L234

Reply:
Some functions even had their tests in-place but the tests were the only usage

Reply:
Yes, I would suggest replacing `strip` by `absl::StripAsciiWhitespace` and `rstrip` by `absl::StripTrailingAsciiWhitespace`.
They also have better names and reduce the technical burden of maintining the `StringUtils` class.

Reply:
I noticed that rstrip is currently taking a non-whitespace argument, so replacing is not possible unfortunately contrary to my initial assumption. I also wasn't able to find an equivalent function in absl

Reply:
How often do we use `rstrip` with specific characters and for what?

Reply:
`rstrip` is used once to trim trailing zeroes, haven't checked the usages of strip though, but I assume it is less than 5

Reply:
`strip` is also only used once, I'm not even sure if this single use makes sense from a semantic standpoint.
If you want we can talk about those 2 remaining usages